### ISSUE TITLE
regarding the metrics function in the model.compile

### ISSUE THREAD

**Issue by surfreta** [fchollet/keras#5375]
According to Keras docuementation, `A metric is a function that is used to judge the performance of your model. Metric functions are to be supplied in the metrics parameter when a model is compiled.`

Is metric used to evaluate the performance only? Will it also affect the training process? I am not very clear about this.

For instance, `model.compile(optimizer=Adam(lr=1e-5), loss='binary_crossentropy', metrics=['binary_crossentropy'])` . I am using `binary_crossentropy `as the objective function for the training process. what are the proper setup for the metrics? Here I just use `binary_crossentropy `as well, can I use other metrics? Does it really matter?

**Comment by unrealwill** [comment]
Hello, as far as I understand metrics are there to display additional information usually "accuracy". They are not driving the gradients. They are mainly useful as a tool to monitor over-fitting. You can decide to use them in heuristics to decide for example when to stop the training. 
You can have a look at : https://github.com/fchollet/keras/blob/master/keras/metrics.py

**Comment by surfreta** [comment]
Hello, thanks.  With respect to monitor over-fitting, I am monitoring the 'val-loss' during the training process.  In other words, when val-loss is improved, the system will automatilly save the model for this epoch; otherwise, not. Is this the right one? How can this interact with monitoring metics to collaborate to overcome over-fitting?  Thanks.

**Comment by unrealwill** [comment]
Yes what you are doing is fine. It depends on what you are interested in and what your problem is. Sometimes you want to to monitor loss, sometimes you want to monitor accuracy. 
Metric are just additional information. You can use them if you want, doesn't mean you have the obligation to.
